### PR TITLE
Ruddy Vincent Refactored Extraordinary Events

### PR DESCRIPTION
**Pull request by regnosys-prod-user**

### CODE REVIEW

**Comment by rvincentISDA** [comment]
The content of this release note should be adequate 
[RELEASE - Part 1.md](https://github.com/finos/common-domain-model/files/12712409/RELEASE.-.Part.1.md)

File: rosetta-source/src/main/rosetta/observable-event-type.rosetta [hunk 1]
```diff
@@ -10,17 +10,17 @@ import cdm.observable.common.*
 
 type AdditionalDisruptionEvents: <"A type for defining the Additional Disruption Events.">
```
Comment (chrisisla) [inline rosetta-source/src/main/rosetta/observable-event-type.rosetta:11]: This type is also in legaldocumentation-master-type now, do we need both?

File: rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta [hunk 2]
```diff
@@ -46,13 +47,93 @@ type MasterAgreementVariableSet: <"Defines a type where additional variables ass
...
+
+    condition InitialStockLoanRate: <" FpML specifies the initialStockLoanRate as a RestrictedPercentage, meaning that its value is comprised between 0 and 1.">
+        if initialStockLoanRate exists then
+            initialStockLoanRate >= 0 and initialStockLoanRate <= 1
+
+    condition DisruptionEventsDeterminingParty:
+        if determiningParty exists then
+            determiningParty = AncillaryRoleEnum -> DisruptionEventsDeterminingParty
+
+type Clause: <"A type for documenting additional clause that cannot yet be represented with the model and yet needed for a digital representation of the agreement">
```
Comment (chrisisla) [inline rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta:134]: This type is very similar to the recently added MasterAgreementClause and associated types. I understand that the changes you are contributing here were done a while ago, before the MasterAgreementClause type was added to the model, but it does look as if there is significant commonality here.

**Comment by chrisisla** [comment]
Hi @rvincentISDA,
If you can take a quick look at the comments and then update the branch from master I'll approve these changes for you.
Thanks!

**Review by rvincentISDA — APPROVED** [review]

File: rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta (hunk 2 above)
Comment (rvincentISDA) [reply rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta:134]: Yes. Clause is the base type that MasterAgreementClause should extend from. MasterAgreementClause is not generic enough to be reusable here. I have shared recommendation for improvement in an email.

Comment (chrisisla) [reply rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta:134]: Clause was not available in the model when MasterAgreementClause was added. I saw your email with suggestions about the Master Agreement types. If you want MasterAgreementClause to extend Clause then I think we will need to investigate and resolve this as part of this PR.

**Review by chrisisla — DISMISSED** [review]
Approved as we will harmonise the structures

File: rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta (hunk 2 above)
Comment (rvincentISDA) [reply rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta:134]: As agreed, we will explore enhancements for masterAgreementClause seperately.

**Review by rvincentISDA — APPROVED** [review]

**Comment by PayalKhanna** [comment]
replaced with https://github.com/finos/common-domain-model/pull/2424

**Comment by PayalKhanna** [comment]
can this be closed please its merged under https://github.com/finos/common-domain-model/pull/2435

**Comment by lolabeis** [comment]
Superseded by #2435

**Pull request closed without merging**